consensus/phases/data_serializers.go

```diff
@@ -0,0 +1,677 @@
+/*
+ *    Copyright 2018 Insolar
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package phases
+
+import (
+	"bytes"
+	"encoding/binary"
+	"io"
+
+	"github.com/insolar/insolar/network/transport"
+	"github.com/pkg/errors"
+)
+
+var defaultByteOrder = binary.BigEndian
+
+type Serializer interface {
+	Serialize() ([]byte, error)
+	Deserialize(data io.Reader) error
+}
+
+// ----------------------------------PHASE 1--------------------------------
+
+// routInfoMasks masks
+const (
+	// take low bit
+	hasRoutingMask = 0x1
+
+	packetTypeMask = 0xf0
+	subtypeMask    = 0xe
+)
+
+func (ph *PacketHeader) parseRouteInfo(routInfo uint8) {
+	ph.PacketT = PacketType((routInfo & packetTypeMask) >> 4)
+	ph.SubType = (routInfo & subtypeMask) >> 1
```

Comment:
define a constant for magic number 1 (something like `packetSubTypeOffset`)

Reply:
done